src/index.d.ts

```diff
@@ -0,0 +1,280 @@
+  getByValue: typeof getByA11yHint
+  getByTestId: typeof getByTestId
+
+  getAllByA11yHint: typeof getAllByA11yHint
+  getAllByA11yLabel: typeof getAllByA11yLabel
+  getAllByA11yRole: typeof getAllByA11yRole
+  getAllByA11yStates: typeof getAllByA11yStates
+  getAllByA11yTraits: typeof getAllByA11yTraits
+  getAllByPlaceholder: typeof getAllByPlaceholder
+  getAllByText: typeof getAllByText
+  getAllByValue: typeof getAllByA11yHint
+  getAllByTestId: typeof getAllByTestId
+
+  queryByA11yHint: typeof queryByA11yHint
+  queryByA11yLabel: typeof queryByA11yLabel
+  queryByA11yRole: typeof queryByA11yRole
+  queryByA11yStates: typeof queryByA11yStates
+  queryByA11yTraits: typeof queryByA11yTraits
+  queryByPlaceholder: typeof queryByPlaceholder
+  queryByText: typeof queryByText
+  queryByValue: typeof queryByA11yHint
+  queryByTestId: typeof queryByTestId
+
+  findByA11yHint: typeof findByA11yHint
+  findByA11yLabel: typeof findByA11yLabel
+  findByA11yRole: typeof findByA11yRole
+  findByA11yStates: typeof findByA11yStates
+  findByA11yTraits: typeof findByA11yTraits
+  findByPlaceholder: typeof findByPlaceholder
+  findByText: typeof findByText
+  findByValue: typeof findByA11yHint
+  findByTestId: typeof findByTestId
+
+  findAllByA11yHint: typeof findAllByA11yHint
+  findAllByA11yLabel: typeof findAllByA11yLabel
+  findAllByA11yRole: typeof findAllByA11yRole
+  findAllByA11yStates: typeof findAllByA11yStates
+  findAllByA11yTraits: typeof findAllByA11yTraits
+  findAllByPlaceholder: typeof findAllByPlaceholder
+  findAllByText: typeof findAllByText
+  findAllByValue: typeof findAllByA11yHint
+  findAllByTestId: typeof findAllByTestId
+}
+
+// QUERY HELPERS
+// -------------
+
+export declare function defaultFilter(node: FiberRoot): boolean
+export declare function getBaseElement(container: ReactTestRenderer | ReactTestInstance): ReactTestInstance
+export declare function getElementError(message: string, container: ReactTestRenderer): Error
+export declare function firstResultOrNull<T extends any[], U>(query: (...args: T) => U[], ...args: T): U | null
+export declare function filterNodeByType(node: FiberRoot, type: string): boolean
+export declare function queryAllByProp(
+  attribute: string,
+  container: ReactTestRenderer,
+  match: TextMatch,
+  options?: NormalizerOptions,
+): FiberRoot[]
+export declare function queryByProp(
+  attribute: string,
+  container: ReactTestRenderer,
+  match: TextMatch,
+  options?: NormalizerOptions,
+): FiberRoot | null
+export function removeBadProperties(node: ReactTestInstance): FiberRoot
+
+// WAIT
+// ----
+
+export interface WaitOptions {
+  timeout?: number
+  interval?: number
+}
+export declare function wait(callback?: () => void, options?: WaitOptions): Promise<void>
+
+// WAIT FOR ELEMENT
+// ----------------
+
+export interface WaitOptions {
+  timeout?: number
+  interval?: number
+}
+export declare function waitForElement<T>(callback: () => T, options?: WaitOptions): Promise<T>
```

Comment:
`waitForElement` also take `container` as an option.

Reply:
This was actually removed, it no longer takes a container 👍 

Unrelated to this PR, but the reason for this was that in `dom-testing-library` the container is used as the root of a `MutationObserver` but we don't actually observe for mutation. It was an option in previous versions, but it didn't actually do anything. This is true for ^1.6.0.